modules/core/src/main/java/org/apache/ignite/internal/processors/service/IgniteServiceProcessor.java

```diff
@@ -679,12 +679,14 @@ private PreparedConfigurations<IgniteUuid> prepareServiceConfigurations(Collecti
             if (err == null) {
                 try {
                     byte[] srvcBytes = U.marshal(marsh, cfg.getService());
+                    byte[] nodeFilterBytes = U.marshal(marsh, cfg.getNodeFilter());
```

Comment:
If we want a bit of optimization, we should check whether `cfg.getNodeFilter()` is not `null`. Otherwise, marshalling doesn't guarantee that a null value is null bytes in the result.